tests/platform_tests/fwutil/fwutil_common.py

```diff
@@ -0,0 +1,260 @@
+    chassis = init_versions["chassis"].keys()[0] # Only one chassis
+    paths = get_install_paths(duthost, fw, init_versions, chassis)
+    current = duthost.shell('sonic_installer list | grep Current | cut -f2 -d " "')['stdout']
+
+    generate_config(duthost, paths, init_versions)
+    upload_platform(duthost, paths, next_image)
+
+    command = "fwutil"
+    if basepath is not None:
+        command += " install"
+    else:
+        command += " update"
+
+    if component is None:
+        command += " all"
+    else:
+        if component not in paths:
+            pytest.skip("No available firmware to install on {}. Skipping".format(component))
+        command += " chassis component {} fw".format(component)
+
+    if basepath is not None:
+        # Install file is override if API implementation needs a different file for install / update
+        filepath = paths[component]["install"]["firmware"] if "install" in paths[component] else paths[component]["firmware"]
+        command += " {}".format(os.path.join(basepath, os.path.basename(filepath)))
+
+    if next_image is not None:
+        command += " --image={}".format("next" if next_image else "current")
+
+    if boot is not None:
+        command += " --boot={}".format(boot)
+
+    command += " -y"
+
+    logger.info("Running install command: {}".format(command))
+    task, res = duthost.command(command, module_ignore_errors=True, module_async=True)
+    boot_type = boot if boot else paths[component]["reboot"][0]
+
+    auto_reboot = False
+    for comp in paths.keys():
+        if "install" in paths[comp] and basepath is not None:
+            if paths[comp]["install"].get("auto_reboot", False): auto_reboot = True
+        else:
+            if paths[comp].get("auto_reboot", False): auto_reboot = True
+
+    timeout = max([v.get("timeout", TIMEOUT) for k, v in paths.items()])
+    pdu_delay = fw["chassis"][chassis].get("power_cycle_delay", 60)
+    complete_install(duthost, localhost, boot_type, res, pdu_ctrl, auto_reboot, current, next_image, timeout, pdu_delay)
+
+    final_versions = show_firmware(duthost)
+    assert validate_versions(init_versions, final_versions, paths, chassis, boot_type)
+
+    duthost.copy(src=os.path.join("firmware", "platform_components_backup.json"), 
+            dest=os.path.join(target, DEVICES_PATH, duthost.facts["platform"], "platform_components.json"))
+    logger.info("Restoring backup platform_components.json to {}".format(
+        os.path.join(DEVICES_PATH, duthost.facts["platform"])))
+
+    update_needed = copy(fw)
+    for comp in paths.keys():
+        if fw["chassis"][chassis]["component"][comp][0]["version"] == final_versions[comp] or paths[comp]["upgrade_only"]:
+            del update_needed["chassis"][chassis]["component"][comp]
+    if len(update_needed["chassis"][chassis]["component"].keys()) > 0:
+        logger.info("Latest firmware not installed after test. Installing....")
```

Comment:
Can we log the component name and boot type as test log even though it can be logged in the device syslog by fwutil?

Reply:
Is this line sufficient? https://github.com/Azure/sonic-mgmt/pull/3814/files/21a024079981fa178d6db7bf7f4dc2f06d2175a1#diff-5f83c4113a2ad4408d2c893d4e5da892ebb136538a83cd4b9aa541871e46ff04R192